ml/backend.go

```diff
@@ -96,6 +96,10 @@ type Tensor interface {
 	Copy(ctx Context, t2 Tensor) Tensor
 }
 
+type ScaledDotProductAttention interface {
```

Comment:
I know that comments aren't in other places here, but we should be documenting these interfaces and functions with comments where this serves as an implementation guide for different back-ends, and a reference for model implementers.

Reply:
I agree, it's good to start documenting these interfaces. Same for the one below.